lib/executor.js

```diff
@@ -316,10 +316,49 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
+    data.config = getUpdatedConfigObject(data.config, app);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
+function getUpdatedConfigObject(config, app) {
+
+  var DYNAMIC_CONFIG_PARAM = /\$\{(\w+)\}$/;
+  var updatedConfigObject = {};
+
+  function getConfigVariable(param) {
+    var configVariable = param;
+    var match = configVariable.match(DYNAMIC_CONFIG_PARAM);
+    if (match) {
+      configVariable = app.get(match[1]);
+      assert(configVariable, param + ' does not resolve to a valid object');
```

Comment:
This is a breaking change - applications that were accidentally using `{...}` values before will stop working now. Please rework the code to emit a warning only and keep the original value. Also make sure that false-y values are not rejected.

```js
if (match) {
  var appValue = app.get(match[1]);
  if (appValue !== undefined) {
    configVariable = appValue;
  } else {
    console.warn(/*...*/);
  }
}
```